src/poetry/core/semver/helpers.py

```diff
@@ -138,10 +144,55 @@ def parse_single_constraint(constraint: str) -> VersionConstraint:
             return VersionRange(min=version)
         if op == ">=":
             return VersionRange(min=version, include_min=True)
+
+        if m.group("wildcard") is not None:
+            return _make_x_constraint_range(
+                version=version,
+                invert=op == "!=",
+                is_marker_constraint=is_marker_constraint,
+            )
+
         if op == "!=":
             return VersionUnion(VersionRange(max=version), VersionRange(min=version))
+
         return version
 
     from poetry.core.semver.exceptions import ParseConstraintError
 
     raise ParseConstraintError(f"Could not parse version constraint: {constraint}")
+
+
+def _make_x_constraint_range(
+    version: Version, invert: bool = False, is_marker_constraint: bool = False
+) -> VersionConstraint:
+    from poetry.core.semver.version_range import VersionRange
+
+    _is_zero_major = version.major == 0 and version.precision == 1
+
+    if version.is_postrelease():
+        _next = version.next_postrelease()
+        _is_zero_major = False
+    elif version.is_stable():
+        _next = version.next_stable()
+    elif version.is_prerelease():
+        _next = version.next_prerelease()
+        _is_zero_major = False
+    elif version.is_devrelease():
+        _next = version.next_devrelease()
+        _is_zero_major = False
+
+    if _is_zero_major:
+        result = VersionRange(max=_next.with_precision(3))
+    else:
+        _min = version.with_precision(max(version.precision, 3))
+
+        if not is_marker_constraint and not _next.is_unstable():
+            _min = _min.next_devrelease()
```

Comment:
Actually, our implementation of `next_devrelease()` is kind of confusing. The next dev release of `1.2.3` can't be `1.2.3dev0` because `1.2.3dev0` was before `1.2.3`. `first_devrelease()` might be a more suitable name.